src/sgame/sg_missile.cpp

```diff
@@ -260,11 +269,18 @@ static int ImpactSlowblob( gentity_t *ent, trace_t *trace, gentity_t *hitEnt )
 	{
 		// extinguish other entity on fire nearby,
 		// and fires on ground
-		if ( neighbor != hitEnt )
+		if ( neighbor != hitEnt && G_IsOnFire( neighbor ) )
 		{
+			reward = true;
 			neighbor->entity->Extinguish( g_abuild_blobFireImmunityDuration.Get() );
 		}
 	}
+	
+	if ( reward )
+	{
+		float credits = g_firefightReward.Get() * CREDITS_PER_EVO;
```

Comment:
Reward is yes or no here. Since a single splat can extinguish several fires, it would be good to have some counting

```suggestion
		float credits = g_firefightReward.Get() * extinguished * CREDITS_PER_EVO;
```

Reply:
Yes, reward is yes or no. This is intended. Also, since you're nitpicking, at least use `unsigned` instead of `int`.

> Since a single splat can extinguish several fires, it would be good to have some counting

I don't think we want to make fire removal that OP credit-wise.

Reply:
I don't think it's a good gameplay mechanic that if there are 2 nearby fires, it's more profitable to extinguish them with 2 separate shots rather than a single shot. That is, I agree with afontain.

Reply:
Oh, this is what you meant. Ok. It makes sense.